### PR TITLE
Feature/error handling just the core formatted

### PR DESCRIPTION
**Pull request by balos1**
Other than applying clang-format and clang-tidy, there are changes/additions in the following files:

- .clang-format
- .clang-tidy
- .github/actions/free-space-ubuntu/action.yml
- .github/workflows/build-llvm-container.yml
- .github/workflows/check-clang-format.yml
- scripts/format.sh
- cmake/SundialsBuildOptionsPre.cmake
- cmake/SundialsSetupCompilers.cmake

Todo after merging:

- [ ] Setup .git-blame-ignore-revs file [so blame still works](https://stackoverflow.com/questions/53502654/how-do-i-run-a-code-formatter-over-my-source-without-modifying-git-history)

### CODE REVIEW

File: scripts/format.sh
Comment (drreynolds) [inline scripts/format.sh:1]: Is this file really supposed to be empty?

Comment (balos1) [reply scripts/format.sh:1]: No, it is not.

**Review by drreynolds — APPROVED** [review]

File: scripts/format.sh
Comment (balos1) [reply scripts/format.sh:1]: I am not sure why GitHub says the file is empty. It is not.

**Review by gardner48 — COMMENTED** [review]
Overall this looks good. Something in the last commit is causing format check action to fail.

**Pull request merged**